src/screens/ChannelListScreenV2/index.tsx

```diff
@@ -1,166 +1,72 @@
 
   React.useEffect(() => {
-    const checkNotif = async () => {
-      if (!navigation.isFocused()) return;
-      PushNotification.checkPermissions(
-        (
-          data: PushNotificationPermissions & {
-            authorizationStatus?: number;
-          }
-        ) => {
-          const lastPromptTime = StorageUtils.lastPromptNotification.get();
-          const promptInterval = 48 * 60 * 60 * 1000;
-          const currentTime = new Date().getTime();
-          if (lastPromptTime && currentTime - parseFloat(lastPromptTime) < promptInterval) {
-            return; // Don't show the prompt if the interval has not passed
-          }
-
-          const needsPermission =
-            data.alert === false || data.badge === false || data.sound === false;
+    if (isFocused) {
+      refresh('channelList');
+      checkNotificationPermission();
+    }
+  }, [isFocused]);
 
-          const showAlert = (onPressAction: () => void) =>
-            Alert.alert(
-              "Don't Miss New Messages",
-              `Allow notifications to know 
-when friends send you messages.`,
-              [
-                {text: 'Not now', onPress: () => console.log('Cancel Pressed')},
-                {text: 'Allow', onPress: onPressAction, isPreferred: true}
-              ]
-            );
+  const renderChannelItem = ({item}) => {
+    if (item?.channelType === 'ANON_PM') {
+      return <MessageChannelItem item={item} onChannelPressed={() => goToChatScreen(item)} />;
+    }
 
-          switch (data.authorizationStatus) {
-            case PERMISSION_STATUS_ACCEPTED:
-              if (needsPermission) {
-                // THIS CASE IT'S TRIGGERED WHEN USER MANUALLY DISABLED THE NOTIFICATION
-                StorageUtils.lastPromptNotification.set(currentTime.toString());
-                showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
-              } else {
-                requestPermission();
-              }
-              break;
-            case PERMISSION_STATUS_BLOCKED:
-              if (needsPermission) {
-                StorageUtils.lastPromptNotification.set(currentTime.toString());
-                showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
-              }
-              break;
-            case PERMISSION_STATUS_PENDING:
-              StorageUtils.lastPromptNotification.set(currentTime.toString());
-              showAlert(() => requestPermission());
-              break;
-            default:
-              break;
-          }
-        }
+    if (item?.channelType === 'ANON_POST_NOTIFICATION') {
+      return (
+        <PostNotificationChannelItem
+          item={item}
+          onChannelPressed={() => goToPostDetailScreen(item)}
+        />
       );
-    };
-
-    checkNotif();
-  }, [navigation?.isFocused?.()]);
+    }
 
-  React.useEffect(() => {
-    if (isFocused) refresh('channelList');
-  }, [isFocused]);
+    return null;
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `renderChannelItem` function checks the `channelType` of each item and renders a different component accordingly. This is a good use of conditional rendering based on the type of the item. However, it might be more maintainable to create a mapping object for the types and their corresponding components. This way, if more types are added in the future, you only need to update the mapping object.

```diff
- const renderChannelItem = ({item}) => {
-   if (item?.channelType === 'ANON_PM') {
-     return <MessageChannelItem item={item} onChannelPressed={() => goToChatScreen(item)} />;
-   }
- 
-   if (item?.channelType === 'ANON_POST_NOTIFICATION') {
-     return (
-       <PostNotificationChannelItem
-         item={item}
-         onChannelPressed={() => goToPostDetailScreen(item)}
-       />
-     );
-   }
- 
-   return null;
- };
+ const CHANNEL_TYPE_COMPONENTS = {
+   'ANON_PM': MessageChannelItem,
+   'ANON_POST_NOTIFICATION': PostNotificationChannelItem
+ };
+
+ const renderChannelItem = ({item}) => {
+   const Component = CHANNEL_TYPE_COMPONENTS[item?.channelType];
+   if (Component) {
+     return <Component item={item} onChannelPressed={() => goToChatScreen(item)} />;
+   }
+   return null;
+ };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->